pkg/client/client_test.go

```diff
@@ -32,16 +33,27 @@ func TestDiscover(t *testing.T) {
 			wantFields: &wantFields{
 				UILocalesSupported: true,
 			},
-			wantErr: false,
+			wantErr: nil,
+		},
+		{
+			name: "discovery failed",
+			args: args{
+				issuer: "https://example.com",
+			},
+			wantFields: &wantFields{
+				UILocalesSupported: true,
+			},
+			wantErr: oidc.ErrDiscoveryFailed,
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			got, err := Discover(context.Background(), tt.args.issuer, http.DefaultClient, tt.args.wellKnownUrl...)
-			if tt.wantErr {
-				assert.Error(t, err)
+			if tt.wantErr != nil {
+				assert.ErrorIs(t, err, tt.wantErr)
 				return
 			}
+
 			require.NoError(t, err)
```

Comment:
Error check can be simplefied:

```suggestion
            require.ErrorIs(t, err, tt.wantErr)
```

Reply:
I understand the suggestion to simplify the error check, but the current implementation uses early returns when an error is encountered to skip the remaining tests. If we simplify the check as proposed, it may cause the tests to fail.

Reply:
When the expectation in `wantFields` is correct (`nil` in this case) the test will not fail.

Reply:
It seems I initially misunderstood your suggestion, but after reconsidering, I’ve incorporated your proposed changes. Thanks!